db/memorystore.go

```diff
@@ -191,6 +200,25 @@ func (m *MemoryStore) GetOrderByID(id string) *core.Order {
 	return nil
 }
 
+func (m *MemoryStore) GetOrdersByAccountID(accountID string) []*core.Order {
+	m.RLock()
+	defer m.RUnlock()
+
+	if orders, ok := m.ordersByAccountID[accountID]; ok {
+		for _, order := range orders {
+			orderStatus, err := order.GetStatus()
+			if err != nil {
+				panic(err)
+			}
+			order.Lock()
+			defer order.Unlock()
+			order.Status = orderStatus
+		}
+		return orders
+	}
+	return make([]*core.Order, 0)
```

Comment:
Is there a reason you preferred returning an empty slice instead of `nil` here?

Reply:
Yes, so that the code calling `db.GetOrdersByAccountID` has a non-`nil` slice and doesn't have to special case for the case it is `nil`.

(It could be of course that this isn't necessary in Go. I'm not that familiar with all the details of Go :) )

Reply:
I think the calling code can be adjusted so that it would handle the `nil` without extra casing. I'll leave comments there.

Reply:
Changed it to return `nil`.